src/Components/Aspire.OpenAI/README.md

```diff
@@ -0,0 +1,123 @@
+
+## Getting started
+
+### Prerequisites
+
+- An OpenAI REST API compatible service like OpenAI.com, Azure AI OpenAI, ollama.com and others.
+- A API key.
+
+### Install the package
+
+Install the .NET Aspire OpenAI library with [NuGet](https://www.nuget.org):
+
+```dotnetcli
+dotnet add package Aspire.OpenAI
+```
+
+## Usage example
+
+In the _Program.cs_ file of your project, call the `AddOpenAIClient` extension method to register an `OpenAIClient` for use via the dependency injection container. The method takes a connection name parameter.
+
+```csharp
+builder.AddOpenAIClient("openaiConnectionName");
+```
+
+You can then retrieve the `OpenAIClient` instance using dependency injection. For example, to retrieve the client from a Web API controller:
+
+```csharp
+private readonly OpenAIClient _client;
+
+public ChatController(OpenAIClient client)
+{
+    _client = client;
+}
+```
+
+To learn how to use the `OpenAIClient` class refer to [Using the OpenAIClient class](https://github.com/openai/openai-dotnet?tab=readme-ov-file#using-the-openaiclient-class).
+
+## Configuration
+
+The .NET Aspire OpenAI library provides multiple options to configure the OpenAI service based on the requirements and conventions of your project. Note that either an `Endpoint` or a `ConnectionString` is required to be supplied.
+
+### Use a connection string
+
+A connection can be constructed from an __Endpoint__ and a __Key__ value with the format `Endpoint={endpoint};Key={key};`. If the `Endpoint` value is omitted the default OpenAI endpoint will be used. You can provide the name of the connection string when calling `builder.AddOpenAIClient()`:
+
+```csharp
+builder.AddOpenAIClient("openaiConnectionName");
+```
+
+#### Connection string
+
+Alternatively, a custom connection string can be used.
+
+```json
+{
+  "ConnectionStrings": {
+    "openaiConnectionName": "Endpoint=https://{openai_rest_api_url}Key={account_key};"
+  }
+}
+```
+
+To obtain an API key on openai.com, first [create a new OpenAI account](https://platform.openai.com/signup) or [log in](https://platform.openai.com/login). 
+Next, navigate to the [API key page](https://platform.openai.com/account/api-keys) and select "Create new secret key", optionally naming the key.
+Make sure to save your API key somewhere safe and do not share it with anyone.
+
+### Use configuration providers
+
+The .NET Aspire OpenAI library supports [Microsoft.Extensions.Configuration](https://learn.microsoft.com/dotnet/api/microsoft.extensions.configuration). It loads the `OpenAISettings` and `OpenAIClientOptions` from configuration by using the `Aspire:OpenAI` key. Example `appsettings.json` that configures some of the options:
+
+```json
+{
+  "Aspire": {
+    "OpenAI": {
+      "DisableTracing": false,
```

Comment:
We should also show how to configure the ClientOptions.

Reply:
If you develop a general-purpose solution for SCM-based clients (which use `ClientPipelineOptions` instead of the Azure.Core `ClientOptions` type), let me know!  We are interested in this piece on our team.

@m-redding, @KrzysztofCwalina  FYI